### PR TITLE
dogecoin: 1.14.2 -> 1.14.3

### PR DESCRIPTION
**Pull request by siraben**
###### Motivation for this change
Much update. Fix darwin build. Minor refactor. Wow.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/blockchains/dogecoin.nix [hunk 1]
```diff
@@ -1,25 +1,22 @@
-{ lib, stdenv , fetchFromGitHub
-, pkg-config, autoreconfHook
-, db5, openssl, boost, zlib, miniupnpc, libevent
-, protobuf, util-linux, qt4, qrencode
-, withGui }:
+{ lib, stdenv, fetchFromGitHub, pkg-config, autoreconfHook, db5
+, openssl, boost, zlib, miniupnpc, libevent, protobuf, qtbase ? null
+, wrapQtAppsHook ? null, qrencode, withGui }:
```
Comment (SuperSandro2000) [inline pkgs/applications/blockchains/dogecoin.nix:3]:
```suggestion
, openssl, boost, zlib, miniupnpc, libevent, protobuf, qtbase
, wrapQtAppsHook, qrencode, withGui }:
```

File: pkgs/applications/blockchains/dogecoin.nix [hunk 2]
```diff
@@ -1,25 +1,22 @@
...
+    sha256 = "sha256-kozUnIislQDtgjeesYHKu4sB1j9juqaWvyax+Lb/0pc=";
   };
 
-  nativeBuildInputs = [ pkg-config autoreconfHook ];
-  buildInputs = [ openssl db5 openssl util-linux
-                  protobuf boost zlib miniupnpc libevent ]
-                  ++ optionals withGui [ qt4 qrencode ];
+  nativeBuildInputs = [ pkg-config autoreconfHook ] ++ optional withGui wrapQtAppsHook;
+  buildInputs = [ openssl db5 openssl protobuf boost zlib miniupnpc libevent ]
+                  ++ optionals withGui [ qtbase qrencode ];
```
Comment (SuperSandro2000) [inline pkgs/applications/blockchains/dogecoin.nix:19]:
```suggestion
  nativeBuildInputs = [ pkg-config autoreconfHook ]
    ++ optional withGui wrapQtAppsHook;
  buildInputs = [ openssl db5 openssl protobuf boost zlib miniupnpc libevent ]
    ++ optionals withGui [ qtbase qrencode ];
```

**Comment by siraben** [comment]
```
error: builder for '/nix/store/zpx928rkkz89yh4z8j3n3vxvy2hkrzbw-dogecoin-1.14.3.drv' failed with exit code 1;
       last 10 log lines:
       > checking for uic-qt5... no
       > checking for uic5... no
       > checking for uic... /nix/store/539h9lcwc1xy17b5wb42xx965nfjkwj4-qtbase-5.12.10-dev/bin/uic
       > checking for rcc-qt5... no
       > checking for rcc5... no
       > checking for rcc... /nix/store/539h9lcwc1xy17b5wb42xx965nfjkwj4-qtbase-5.12.10-dev/bin/rcc
       > checking for lrelease-qt5... no
       > checking for lrelease5... no
       > checking for lrelease... no
       > configure: error: LRELEASE not found
       For full logs, run 'nix log /nix/store/zpx928rkkz89yh4z8j3n3vxvy2hkrzbw-dogecoin-1.14.3.drv'.
```

**Comment by siraben** [comment]
Ok it's actually building now.

**Comment by RaghavSood** [comment]
Result of `nixpkgs-review pr 114769` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>dogecoind</li>
  </ul>
</details>

**Review by RaghavSood — APPROVED** [review]

File: pkgs/applications/blockchains/dogecoin.nix [hunk 3]
```diff
@@ -1,29 +1,32 @@
-{ lib, stdenv , fetchFromGitHub
-, pkg-config, autoreconfHook
-, db5, openssl, boost, zlib, miniupnpc, libevent
-, protobuf, util-linux, qt4, qrencode
-, withGui }:
+{ lib, stdenv, fetchFromGitHub, pkg-config, autoreconfHook, db5
+, openssl, boost, zlib, miniupnpc, libevent, protobuf, qtbase ? null
+, wrapQtAppsHook ? null, qttools ? null, qmake ? null, qrencode, withGui
+, hexdump, Cocoa ? null }:
```
Comment (SuperSandro2000) [inline pkgs/applications/blockchains/dogecoin.nix:4]:
```suggestion
, wrapQtAppsHook, qttools, qmake, qrencode, withGui
, hexdump, Cocoa }:
```

That should be required if it is behind a with flag.

File: pkgs/applications/blockchains/dogecoin.nix [hunk 4]
```diff
@@ -1,29 +1,32 @@
...
 
 with lib;
```
Comment (SuperSandro2000) [inline pkgs/applications/blockchains/dogecoin.nix:6]: That could be removed to make it clear where lib is used.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Review by lblasc — APPROVED** [review]

**Comment by lblasc** [comment]
Much works. Please merge. Amaze.

**Comment by prusnak** [comment]
Another PR (which implements things not included in this PR) https://github.com/NixOS/nixpkgs/pull/121941

**Comment by siraben** [comment]
Looks like I can incorporate those changes here.

**Comment by centromere** [comment]
> Looks like I can incorporate those changes here.

That would be very helpful. Thank you @siraben!

**Comment by ajs124** [comment]
This has conflicts + we're on a newer version already. Can we close this?

**Pull request closed without merging**